runtime/s390x.S

```diff
@@ -693,10 +693,10 @@ CFI_STARTPROC
     /* Initial shuffling of arguments */
     /* (%r2 = Caml_state, %r3 = closure, 0(%r4) = arg1, 8(%r4) = arg2) */
         lgr     TMP3, C_ARG_1       /* Caml_state */
-        lgr     TMP2, C_ARG_2
+        lgr     C_ARG_4, C_ARG_2
         lg      C_ARG_1, 0(C_ARG_3) /* first argument */
         lg      C_ARG_2, 8(C_ARG_3) /* second argument */
-        lgr     C_ARG_3, TMP2       /* closure */
+        lgr     C_ARG_3, C_ARG_4    /* closure */
```

Comment:
Noob question, why not use TMP here? (It is set below, but that is not obviously a problem.)

Reply:
Using `TMP` would work as well. Using `C_ARG_4` as done in the PR is closer to what `caml_callback3_asm` does, which should make reviewer more confident that this change is correct.

Reply:
The difference is that, for caml_callback3_asm, C_ARG_4 is the intended final location of the argument, while here it is only a temporary location to avoid writing to C_ARG_3 too early. Personally I would find TMP more readable, but I am now convinced that either versions are correct.

Reply:
`TMP` it is, then.